rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2042,7 +2042,13 @@ extern "C" rmw_ret_t rmw_publisher_count_matched_subscriptions(
   const rmw_publisher_t * publisher,
   size_t * subscription_count)
 {
-  RET_WRONG_IMPLID(publisher);
```

Comment:
While this is different from the other RMW implementations, I'm thinking we should probably keep using `RET_WRONG_IMPLID` and change the return value of the macro to `RMW_RET_INCORRECT_RMW_IMPLEMENTATION`.

That is, this rmw uses `RET_WRONG_IMPLID` in a lot of places, and now we'll have different return values at different entry points.

What do you think?

Reply:
I agree with changing `RET_WRONG_IMPLID` to return `RMW_RET_INCORRECT_RMW_IMPLEMENTATION`, but I'm somewhat inclined towards standardizing how such errors are handled. Why having two macros to do the same thing?

Reply:
> I agree with changing `RET_WRONG_IMPLID` to return `RMW_RET_INCORRECT_RMW_IMPLEMENTATION`, but I'm somewhat inclined towards standardizing how such errors are handled. Why having two macros to do the same thing?

I agree, but I think it is more important how the API reacts than the internal implementation details.  I'm still worried that we are only changing part of the API to return a different value.  With that in mind, my suggestion here is that this PR updates `RET_WRONG_IMPLID` to return `RMW_RET_INCORRECT_RMW_IMPLEMENTATION`, but leaves this call to `RET_WRONG_IMPLID` in place.  Then in a follow-up we can replace all uses of `RET_WRONG_IMPLID` with `RMW_CHECK_TYPE_IDENTIFIERS_MATCH`.  Thoughts about that?

Reply:
I'd rather keep this PR small and in scope (i.e. I'd rather not touch the return code of every function that uses `RET_WRONG_IMPLID` here). 

What if we put this one on hold, phase out `RET_WRONG_IMPLID` in a separate PR, and then rebase this one on top?

Reply:
Alright, ended up changing `RET_WRONG_IMPLID` only. See https://github.com/ros2/rmw_cyclonedds/pull/226. It was the simplest change, as the implementation isn't consistent about what to return when an argument is `nullptr` either. 

I'll standardize it over time as I keep adding tests, but I want to avoid blanket PRs now.